test/e2e/antreapolicy_test.go

```diff
@@ -655,50 +655,36 @@ func testANPBasic(t *testing.T) {
 }
 
 // testAuditLoggingBasic tests that a audit log is generated when egress drop applied
-func testAuditLoggingBasic(t *testing.T) {
+func testAuditLoggingBasic(t *testing.T, data *TestData) {
 	builder := &ClusterNetworkPolicySpecBuilder{}
 	builder = builder.SetName("test-log-cnp-deny").
 		SetPriority(1.0).
-		SetAppliedToGroup(map[string]string{"pod": "a"}, nil, nil, nil)
+		SetAppliedToGroup(map[string]string{"pod": "a"}, map[string]string{"ns": "x"}, nil, nil)
 	builder.AddEgress(v1.ProtocolTCP, &p80, nil, nil, nil, map[string]string{"ns": "z"},
-		nil, nil, secv1alpha1.RuleActionDrop).
-		AddEgressLogging()
+		nil, nil, secv1alpha1.RuleActionDrop)
+	builder.AddEgressLogging()
 
-	reachability := NewReachability(allPods, true)
-	reachability.Expect(Pod("x/a"), Pod("z/a"), false)
-	reachability.Expect(Pod("x/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("x/a"), Pod("z/c"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/a"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/c"), false)
-	reachability.Expect(Pod("z/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("z/a"), Pod("z/c"), false)
+	_, err := k8sUtils.CreateOrUpdateCNP(builder.Get())
+	failOnError(err, t)
 
-	testStep := []*TestStep{
-		{
-			"Port 80",
-			reachability,
-			[]metav1.Object{builder.Get()},
-			80,
-			0,
-		},
-	}
-	testCase := []*TestCase{
-		{"Audit Log CNP Drop Egress From All Pod:a to NS:z", testStep},
-	}
-	executeTests(t, testCase)
-}
+	// generate some traffic that will be dropped by test-log-cnp-deny
+	k8sUtils.Probe("x", "a", "z", "a", p80)
+	k8sUtils.Probe("x", "a", "z", "b", p80)
+	k8sUtils.Probe("x", "a", "z", "c", p80)
 
-// validateAuditLog validates whether testAuditLoggingBasic succeeded or failed
-func validateAuditLog() bool {
-	data, err := ioutil.ReadFile("/var/log/antrea/networkpolicy/np.log")
+	podXA, _ := k8sUtils.GetPod("x", "a")
+	nodeName := podXA.Spec.NodeName
```

Comment:
maybe add a quick comment to say that `NodeName` is guaranteed to be set at this stage since the framework waits for all the Pods to be in the `Running` phase

Reply:
Done. Thanks for the suggestion.